src/scalar.jl

```diff
@@ -52,7 +53,144 @@ transform(::Identity, x::Real) = x
+function inverse(::TVExp, x::Number)
+    @argcheck x > 0 DomainError
+    log(x)
+end
+inverse_and_logjac(t::TVExp, x::Number) = inverse(t, x), log(1/x)
+
+"""
+$(TYPEDEF)
+
+Logistic transformation `x ↦ logit(x)`. Maps from all reals to (0, 1).
+"""
+struct TVLogistic <: ScalarTransform
+end
+transform(::TVLogistic, x::Real) = logistic(x)
+transform_and_logjac(t::TVLogistic, x::Real) = transform(t, x), logistic_logjac(x)
+
+function inverse(::TVLogistic, x::Number)
+    @argcheck 0 < x < 1 DomainError
+    logit(x)
+end
+inverse_and_logjac(t::TVLogistic, x::Number) = inverse(t, x), logit_logjac(x)
+
+"""
+$(TYPEDEF)
+
+Shift transformation `x ↦ x + shift`. 
+"""
+struct TVShift{T <: Real} <: ScalarTransform
+    shift::T
+end
+transform(t::TVShift, x::Real) = x + t.shift
+transform_and_logjac(t::TVShift, x::Real) = transform(t, x), zero(x)
+
+inverse(t::TVShift, x::Number) = x - t.shift
+inverse_and_logjac(t::TVShift, x::Number) = inverse(t, x), zero(x)
+
+"""
+$(TYPEDEF)
+
+Scale transformation `x ↦ scale * x`.
+"""
+struct TVScale{T} <: ScalarTransform
+    scale::T
+    function TVScale{T}(scale::T) where {T}
+        @argcheck scale > zero(scale) DomainError
+        new(scale)
+    end
+end
+TVScale(scale::T) where {T} = TVScale{T}(scale)
+
+transform(t::TVScale, x::Real) = t.scale * x
+transform_and_logjac(t::TVScale{T}, x::Real) where {T<:Real} = transform(t, x), log(t.scale) 
+
+inverse(t::TVScale, x::Number) = x / t.scale
+inverse_and_logjac(t::TVScale{T}, x::Number) where {T<:Real} = inverse(t, x), -log(t.scale)
+
+"""
+$(TYPEDEF)
+
+Negative transformation `x ↦ -x`.
+"""
+struct TVNeg <: ScalarTransform
+end
+
+transform(::TVNeg, x::Real) = -x
+transform_and_logjac(t::TVNeg, x::Real) = transform(t, x), zero(x)
+
+inverse(::TVNeg, x::Number) = -x
+inverse_and_logjac(::TVNeg, x::Number) = -x, zero(x)
+
+####
+#### composite scalar transforms
+####
+"""
+$(TYPEDEF)
+
+A composite scalar transformation, i.e. a sequence of scalar transformations.
+The component transforms can be accessed indexing into the `CompositeScalarTransform`, 
+as in `ct[i]` or `ct[end]`.
+"""
+struct CompositeScalarTransform{Ts <: Tuple} <: ScalarTransform
+    transforms::Ts
+    function CompositeScalarTransform(transforms::Ts) where {Ts <: Tuple}
+        @argcheck length(transforms) > 0 DomainError
+        @argcheck all(t -> t isa ScalarTransform, transforms) DomainError
+        new{Ts}(transforms)
+    end
+end
+
+Base.getindex(t::CompositeScalarTransform, i) = t.transforms[i]
+Base.firstindex(t::CompositeScalarTransform) = firstindex(t.transforms)
+Base.lastindex(t::CompositeScalarTransform) = lastindex(t.transforms)
+
+transform(t::CompositeScalarTransform, x) = foldr((t, y) -> transform(t, y), t.transforms, init=x)
+function transform_and_logjac(ts::CompositeScalarTransform, x) 
+    foldr(ts.transforms, init=(x, zero(x))) do t, (x, logjac)
+        nx, nlogjac = transform_and_logjac(t, x)
+        x = nx
+        logjac += nlogjac
+        (x, logjac)
```

Comment:
IMO a bit confusing to reuse the function argument names here:
```suggestion
        (nx, logjac + nlogjac)
```